sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala

```diff
@@ -29,22 +29,9 @@ import org.apache.spark.sql.catalyst.rules.Rule
 import org.apache.spark.sql.execution.datasources.DataSourceStrategy
 import org.apache.spark.sql.internal.SQLConf
 
-/**
- * Prune hive table partitions using partition filters on [[HiveTableRelation]]. The pruned
- * partitions will be kept in [[HiveTableRelation.prunedPartitions]], and the statistics of
- * the hive table relation will be updated based on pruned partitions.
- *
- * This rule is executed in optimization phase, so the statistics can be updated before physical
- * planning, which is useful for some spark strategy, eg.
- * [[org.apache.spark.sql.execution.SparkStrategies.JoinSelection]].
- *
- * TODO: merge this with PruneFileSourcePartitions after we completely make hive as a data source.
- */
-private[sql] class PruneHiveTablePartitions(session: SparkSession)
```

Comment:
Why remove these comments?

Reply:
When i change from class to object, the building task java unidoc failed。Do you know how to fix this issue??

Reply:
-Phadoop-3.2 -Phive-2.3 -Phadoop-cloud -Pkinesis-asl -Pspark-ganglia-lgpl -Pyarn -Pkubernetes -Phive -Pmesos -Phive-thriftserver unidoc

Reply:
I will upload the error log later to see if you can help me on this issue.

Reply:
https://github.com/akka/akka-http/issues/1895
something might be relative??

Reply:
error message as following

```
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:3:1:  错误: 找不到引用
[error]  * Prune hive table partitions using partition filters on {@link HiveTableRelation}. The pruned
[error]                                                                  ^
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:4:1:  错误: 找不到引用
[error]  * partitions will be kept in {@link HiveTableRelation.prunedPartitions}, and the statistics of
[error]                                      ^
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:9:1:  错误: 找不到引用
[error]  * {@link org.apache.spark.sql.execution.SparkStrategies.JoinSelection}.
[error]           ^
```

Reply:
I was trying to add a empty class definition before object, the unidoc build passed.

```
private[sql] class PruneHiveTablePartitions

/**
 * Prune hive table partitions using partition filters on [[HiveTableRelation]]. The pruned
 * partitions will be kept in [[HiveTableRelation.prunedPartitions]], and the statistics of
 * the hive table relation will be updated based on pruned partitions.
 *
 * This rule is executed in optimization phase, so the statistics can be updated before physical
 * planning, which is useful for some spark strategy, eg.
 * [[org.apache.spark.sql.execution.SparkStrategies.JoinSelection]].
 *
 * TODO: merge this with PruneFileSourcePartitions after we completely make hive as a data source.
 */
private[sql] object PruneHiveTablePartitions
  extends Rule[LogicalPlan] with CastSupport with PredicateHelper {
```

Reply:
So, how do I fix this issue?
1. remove the comment
2. remove the comment with [[]] link
3. add a empty class definition

Reply:
@cloud-fan FYI. need your suggestion too.

Reply:
we can replace [[]] with backticks
```
... using partition filters on `HiveTableRelation`.
```